lib/src/features/main/theme/theme_data/app_theme_data.dart

```diff
@@ -0,0 +1,283 @@
+import 'package:flutter/material.dart';
+import 'package:gui/gen/fonts.gen.dart';
+import 'package:gui/src/core/enums/theme_modes.dart';
+import 'package:gui/src/features/main/theme/theme_data/pallets/on_surface_pallet.dart';
+import 'package:gui/src/features/main/theme/theme_data/pallets/surface_pallet.dart';
+
+/// ## [AppThemeData] Documentation
+///
+/// The [AppThemeData] class provides theming configurations for the
+/// application.
+/// It contains a static map that holds the `ThemeData` for both light and dark
+/// modes, allowing for easy theme management based on the user's preference.
+///
+class AppThemeData {
+  const AppThemeData._();
+  static final Map<ThemeModes, ThemeData> themeDataModes = {
+    // light ThemeData.
+    ThemeModes.light: ThemeData(
+      bottomNavigationBarTheme: BottomNavigationBarThemeData(
+        backgroundColor: SurfacePallet.light.surface3,
+      ),
+      scaffoldBackgroundColor: SurfacePallet.light.surface3,
+      brightness: Brightness.light,
+      dividerColor: OnSurfacePallet.light.onSurface1,
+      textTheme: lightTextTheme,
+      cardColor: SurfacePallet.light.surface3,
+      extensions: const <ThemeExtension<dynamic>>[
+        OnSurfacePallet.light,
+        SurfacePallet.light,
+      ],
+    ),
+
+    // dark ThemeData.
+    ThemeModes.dark: ThemeData(
+      brightness: Brightness.dark,
+      textTheme: darkTextTheme,
+      extensions: <ThemeExtension<dynamic>>[
+        OnSurfacePallet.dark,
+        SurfacePallet.dark,
+      ],
+    ),
+  };
+
+  static final lightTextTheme = TextTheme(
+    displaySmall: TextStyle(
+      fontFamily: FontFamily.inter,
+      color: SurfacePallet.light.surface3,
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.w700,
+      fontSize: 16,
+      height: 24 / 12,
+      leadingDistribution: TextLeadingDistribution.even,
+    ),
+    headlineLarge: TextStyle(
+      fontFamily: FontFamily.inter,
+      color: SurfacePallet.light.surface3,
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.w300,
+      fontSize: 36,
+      height: 52 / 36,
+      leadingDistribution: TextLeadingDistribution.even,
+    ),
+    headlineMedium: TextStyle(
+      fontFamily: FontFamily.inter,
+      color: SurfacePallet.light.surface3,
+      fontStyle: FontStyle.normal,
+      fontWeight: FontWeight.w400,
+      fontSize: 32,
```

Comment:
Please consider defining the font size as a constant 

  static const fontSizes = FontSizes(
    displaySmall: 16,
    headlineLarge: 36,
    // ... other sizes
  );

Reply:
If all the issues are related to text styles, we already have a dedicated issue addressing this.
We can proceed with finishing this merge request.

----
@phoenixit99

Reply:
Ok.got it We can merge request the PR